mmpose/datasets/pipelines/shared_transform.py

```diff
@@ -140,3 +141,94 @@ def __repr__(self):
         """Compute the string representation."""
         return (f'{self.__class__.__name__}('
                 f'keys={self.keys}, meta_keys={self.meta_keys})')
+
+
+@PIPELINES.register_module()
+class HideAndSeek:
+    """Augmentation by informantion dropping in Hide-and-Seek paradigm. Paper
+    ref: Huang et al. AID: Pushing the Performance Boundary of Human Pose
+    Estimation with Information Dropping Augmentation (arXiv:2008.07139 2020).
+
+    Args:
+        prob_has (float): Probability of performing hide-and-seek.
+        prob_has_hide (float): Probability of hiding patches.
+        grid_sizes (list): List of optional grid sizes.
+    """
+
+    def __init__(self,
+                 prob_has=1.0,
+                 prob_has_hide=0.5,
+                 grid_sizes=(0, 16, 32, 44, 56)):
+        self.prob_has = prob_has
+        self.prob_has_hide = prob_has_hide
+        self.grid_sizes = grid_sizes
+
+    def _hide_and_seek(self, img):
+        # get width and height of the image
+        height, width, _ = img.shape
+
+        # randomly choose one grid size
+        index = np.random.randint(0, len(self.grid_sizes) - 1)
+        grid_size = self.grid_sizes[index]
+
+        # hide the patches
+        if grid_size != 0:
+            for x in range(0, width, grid_size):
+                for y in range(0, height, grid_size):
+                    x_end = min(width, x + grid_size)
+                    y_end = min(height, y + grid_size)
+                    if np.random.rand() <= self.prob_has_hide:
+                        img[x:x_end, y:y_end, :] = 0
+        return img
+
+    def __call__(self, results):
+        img = results['img']
+        if np.random.rand() < self.prob_has:
+            img = self._hide_and_seek(img)
+        results['img'] = img
+        return results
+
+
+@PIPELINES.register_module()
+class Cutout:
+    """Augmentation by informantion dropping in Cutout paradigm. Paper ref:
```

Comment:
To help future readers and users, please provide a desription on difference between this implementation and the original paper's version, and also add reference to the original paper.